frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -845,7 +885,24 @@ pub mod pallet {
             PalletDisabled::<T>::put(enable_maintenance);
 
             Self::deposit_event(Event::<T>::MaintenanceMode(enable_maintenance));
+            Ok(().into())
+        }
 
+        /// `StakeBalance` (default) means claimed amount gets restaked,
+        /// `FreeBalance` sets claiming without restaking
+        #[pallet::weight(T::WeightInfo::set_reward_destination())]
+        pub fn set_reward_destination(
+            origin: OriginFor<T>,
+            reward_destination: RewardDestination,
+        ) -> DispatchResultWithPostInfo {
+            Self::ensure_pallet_enabled()?;
+            let staker = ensure_signed(origin)?;
+
+            Ledger::<T>::mutate(&staker, |ledger| {
+                ledger.reward_destination = reward_destination;
```

Comment:
Just noticed that this will leave a DB entry even if an account isn't staking.
This shouldn't be the case since we cleanup empty ledgers.

You should ensure that the ledger isn't empty before mutating it.
Add a UT for the scenario too.

Reply:
good catch 👍